### PR TITLE
test for hash functions and ip types

### PR DESCRIPTION
**Pull request by den-crane**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

test for https://github.com/ClickHouse/ClickHouse/pull/45024

### CODE REVIEW

**Comment by den-crane** [comment]
```sql
22.12

select intHash64(toIPv4('1.2.3.4'));
--
5715546585361069049

22.13
Code: 43. DB::Exception: Illegal type IPv4 of argument of function intHash64: While processing intHash64(toIPv4('1.2.3.4')). (ILLEGAL_TYPE_OF_ARGUMENT)
```

The same for 
`intHash32`
`jumpConsistentHash`
`kostikConsistentHash`
`yandexConsistentHash`

**Comment by den-crane** [comment]
`select intHash64(toIPv4('1.2.3.4'));` -- was undefined behaviour because `intHash64` accepts `int` not `ip` types.
I can fix the test.

**Comment by yakov-olkhovskiy** [comment]
probably it makes sense to accept IPv4 for intHash?

**Comment by den-crane** [comment]
> probably it makes sense to accept IPv4 for intHash?

Up to you. 
There is a tiny chance that somebody uses it as a sharding key.

@alexey-milovidov

**Comment by den-crane** [comment]
Also 

```
select toIPv4('1.2.3.4') % 10;

select intDiv(toIPv4('1.2.3.4'), 10);

```

Some "smart" can use it like this

```
create table A ( ip IPv4) engine MergeTree partition by ip % 10 order by tuple();
```

**Comment by alexey-milovidov** [comment]
Yes, let's add it.

**Comment by yakov-olkhovskiy** [comment]
https://github.com/ClickHouse/ClickHouse/pull/45043
it seems strange but `farmHash64` for strings changed, so test for IPv6 fails... I'm not sure yet if it was from my initial PR for IP types - don't see how it could be related...

**Comment by yakov-olkhovskiy** [comment]
wow, this is very strange - `farmHash64` gives different results for IPv6 (and, apparently for any string) on my local build than on CI... I have no idea what's going on...

**Comment by yakov-olkhovskiy** [comment]
my local build:
```
ip-172-31-42-195.us-east-2.compute.internal :) SELECT version(), farmHash64('ddd')

SELECT
    version(),
    farmHash64('ddd')

Query id: 4a31b490-d492-47ff-93a4-1033fa94f365

┌─version()─┬────farmHash64('ddd')─┐
│ 22.13.1.1 │ 18136076161938598894 │
└───────────┴──────────────────────┘

1 row in set. Elapsed: 0.006 sec.
```
CI build:
```ip-172-31-42-195.us-east-2.compute.internal :) SELECT version(), farmHash64('ddd')

SELECT
    version(),
    farmHash64('ddd')

Query id: f97f3f19-b3ce-4b21-826f-0d84af6989a4

┌─version()─┬────farmHash64('ddd')─┐
│ 22.13.1.1 │ 12114677674100768748 │
└───────────┴──────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```

**Comment by alexey-milovidov** [comment]
FarmHash is tricky:
- the function was created specifically to be unstable and not intended to be used for storage, partitioning, etc;
- but we added it to ClickHouse, because it was already used in another system in the company;
- then we have to fixate its behavior, so it will never change;
- maybe we failed to fixate its behavior, and it is platform-dependent?
- then we need to make it behave exactly as on x86_64 machine with maximum SSE 4.2.

**Comment by yakov-olkhovskiy** [comment]
> Some "smart" users can use it like this

for this I think I will need to add binary arithmetic for IPv4 - will do it in different PR

File: tests/queries/0_stateless/02184_hash_functions_and_ip_types.sql [hunk 1]
```diff
@@ -0,0 +1,44 @@
+SELECT
```
Comment (rschu1ze) [inline tests/queries/0_stateless/02184_hash_functions_and_ip_types.sql:1]: BLAKE, MD*() and SHA*() are missing. Intended?

**Comment by rschu1ze** [comment]
I see that this PR was integrated into #45040. I think this PR can be closed then.

**Comment by yakov-olkhovskiy** [comment]
I added this test into https://github.com/ClickHouse/ClickHouse/pull/45043
closing

**Pull request closed without merging**